### PR TITLE
Rewrite a bunch of commands to Typescript

### PR DESCRIPTION
**Pull request by bkendall**
### Description

Commands updated:

- `auth:export`
- `functions:config:clone`
- `functions:config:export`
- `functions:config:get`
- `functions:config:legacy` is removed - it's not used anywhere
- `functions:config:set`
- `functions:config:unset`
- `help`
- `login:ci`
- `setup:emulators:database`
- `setup:emulators:firestore`
- `setup:emulators:ui`
- `target:apply`
- `target:clear`
- `target:remove`
- `target`

### Scenarios Tested

- [x] `auth:export`
- [x] `functions:config:clone`
- [x] `functions:config:export`
- [x] `functions:config:get`
- [x] `functions:config:legacy` is removed - it's not used anywhere
- [x] `functions:config:set`
- [x] `functions:config:unset`
- [x] `help`
- [x] `login:ci`
- [x] `setup:emulators:database`
- [x] `setup:emulators:firestore`
- [x] `setup:emulators:ui`
- [x] `target:apply`
- [x] `target:clear`
- [x] `target:remove`
- [x] `target`

### CODE REVIEW

File: src/commands/functions-config-unset.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "runtimeconfig.variables.list",
+    "runtimeconfig.variables.create",
+    "runtimeconfig.variables.get",
+    "runtimeconfig.variables.update",
+    "runtimeconfig.variables.delete",
+  ])
+  .before(functionsConfig.ensureApi)
+  .action(async (args, options) => {
+    if (!args.length) {
+      return utils.reject("Must supply at least one key");
```
Comment (taeold) [inline src/commands/functions-config-unset.ts:28]: nit* Don't really care, but I'm guessing this could also be `new FirebaseError`

**Review by taeold — APPROVED** [review]
What a machine.

File: src/commands/target.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      const targets = options.rc.targets(options.project, type);
+      logTargets(type, targets);
+      return targets;
+    }
+
+    const allTargets: { [product: string]: targetMap } = options.rc.allTargets(options.project);
+    for (const [targetType, targetName] of Object.entries(allTargets)) {
+      logTargets(targetType, targetName);
+    }
+    return allTargets;
```
[the remark this answers is not in the document]
Comment (bkendall) [reply src/commands/target.ts:39]: In this case, yes. Nothing here is async and it's not strictly necessary to return a Promise to `action`

File: src/commands/functions-config-unset.ts (hunk 1 above)
Comment (bkendall) [reply src/commands/functions-config-unset.ts:28]: Done.

**Pull request merged**